framework/src/actions/SetupMeshAction.C

```diff
@@ -205,17 +205,17 @@ std::string
 SetupMeshAction::modifyParamsForUseSplit(InputParameters & moose_object_params) const
 {
   // Get the split_file extension, if there is one, and use that to decide
-  // between .cpr and .cpa
+  // between .cpr and .cpa.gz
   auto split_file = _split_file;
   std::string split_file_ext;
   auto pos = split_file.rfind(".");
   if (pos != std::string::npos)
     split_file_ext = split_file.substr(pos + 1, std::string::npos);
```

Comment:
Wait, how is this still going to find the full extension if the extension is ".cpa.gz"?  Won't the `rfind(".")` stop at the second "." and just leave `split_file_ext == "gz"`, which would then fail to hit the code path we expect below?

Reply:
Good catch. I'm surprised nothing in the test suite caught this. I'll add a test.

Reply:
Right!?  If I'd been more confident and less confused I'd have requested such a test myself, but I still had some hope that you'd come back with, "No, what you're missing is that this code still works because ..."

Reply:
I like your solution here; it has the issue of false positives in the *other* direction (directories with `.` in the name, say), but I don't know how we'd avoid that without a predefined list of possible suffixes or a short maximum suffix length or something.

Reply:
We could always do an rfind("/") to detect any folders prepended to the file name and split off the file part. But that's an issue for future moose developers, not current moose developers :)